module-4/backend/week_four.md

```diff
@@ -5,15 +5,75 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's your favorite project management tool?
+  - Personally, I like Waffle.io for the simplicity. But I see why Pivitol is used on by companies. It is a good one stop shop for everyone to be aware of what is being worked on. 
 2. Why do we create staging environments?
+  - So that we can create environments as close to different production environments as possible before deploying.
 3. What are the characteristics of a good README (in your opinion)?
+  - References to contributers
+  - Setup Instructions
+    - Links to external keys/packages/etc. needed
+  - How to Run/Use
+  - Features you wish to highlight
+  - Instructions on contribution (if offered)
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+  - Being aware of color-blindness and dyslexia when choosing colors/fonts for CSS/SASS
+  - Utilization of buttons instead of links for people who need to use `tab` instead of a mouse
 5. What are some basic security concerns to be aware of when building applications?
+  - If storing passwords, salt
+  - Don't publish production keys, tokens, etc.
+  - Make sure ORM protects against SQL injection (utilize ORM instead of raw SQL if possible), or sanitize information coming from params manually before passing to SQL query,
 6. Why is continuous integration helpful/important?
+  - It's an additional level of testing that we can introduce to our program including running with different versions of different languages to help indentify bugs. 
 7. What are some continuous integration tools?
+  - TravisCI, Jenkins, CircleCI
 
 #### Review  
 
 8. What are some characteristics of "good" git workflow?
+  - Checking out branches based on feature/task/bug/etc and creating PRs for review when complete
+  - Not merging your own PR
+  - Multiple comments attached to code snippets when reviewing PRs & issues
+  - `@`ing other contributers to alert them when they are ready for review
+  - Summary when submitting something for review, breakdown of key parts, requests for help specifically
```

Comment:
i love all of these bullet points - i'd love for you to dive deeper into the "why" behind each of them though to demonstrate a deeper level of understanding rather than just simply answering the question. why shouldn't we merge our own PR? why do we want to even checkout branches and make PR's? why not just work on master?